Paws4Me/Paws4Me/Controllers/Logic/ViewController.swift

```diff
@@ -8,109 +8,243 @@
+           return indexRow
+       }
+
+       override func prepare(for segue: UIStoryboardSegue, sender: Any?) {
+           if let destination = segue.destination as? PetSingleDetailsViewController {
+               let indexRow = getIndexPetSelected()
+               guard let pageItem = adoptPetObject?.page?[indexRow] else { return }
+               if let pagePetName = pageItem.name {
+                   destination.namePet = pagePetName
+               }
+               if let pagePetAge = pageItem.age {
+                   destination.agePet = pagePetAge
+               }
+               if let pageGender = pageItem.sex {
+                   destination.genderPet = pageGender
+               }
+               if let pagePetImage = pageItem.animalImage {
+                   destination.imgPet = pagePetImage
+               }
+               if let pagePetBreed = pageItem.animalSpeciesBreed?.petBreedName {
+                   destination.breedPet = pagePetBreed
+               }
+           }
+       }
+
+       // MARK: - Search
+       func searchBar(_ searchBar: UISearchBar, selectedScopeButtonIndexDidChange selectedScope: Int) {
+           if selectedScope == 0 {
+               selectedGender = "male"
+           } else if selectedScope == 1 {
+               selectedGender = "female"
+           }
+           searchPetType(type: petType)
+       }
+
+       func searchBar(_ searchBar: UISearchBar, textDidChange searchText: String) {
+           if searchText == "" {
+               filteredPetObject = adoptPetObject
+           } else {
+               isSingleSearch = true
+               guard let petObject = adoptPetObject else {
+                   return
+               }
+               let searchText = searchText.lowercased()
+               filteredPetObject?.page = petObject.page?.filter { $0.name?.lowercased().starts(with: searchText)
+                   ??  false}
+               indexSinglePet =  petObject.page?.firstIndex(
+                where: { $0.name?.lowercased().starts(with: searchText) ??  false })
+               print(filteredPetObject as Any)
+               if filteredPetObject == nil {
+                   filteredPetObject = adoptPetObject
+               }
+           }
+           self.petTable.reloadData()
+       }
+
+       func searchPetType(type: String) {
+           if type == "" && selectedGender == ""{
+               filteredPetObject = adoptPetObject
+           } else {
+               guard adoptPetObject != nil else { return }
+               let petAge = getAgeFromType(type: type)
+               searchPet(type: type, gender: selectedGender, petAge: petAge)
+               if filteredPetObject == nil {
+                   filteredPetObject = adoptPetObject
+               }
+                   self.petTable.reloadData()
+               }
+           }
+
+       func getAgeFromType(type: String) -> String {
+           var petAge = ""
+           if type == "kitten" || type == "puppy" {
```

Comment:
Start looking at enums for different types instead of strings. What if "Puppy" was passed in it could be missed